packages/http-client-csharp/generator/TestProjects/Local/Unbranded-TypeSpec/src/Generated/Internal/Optional.cs

```diff
@@ -0,0 +1,55 @@
+// <auto-generated/>
+
+#nullable disable
+
+using System.Collections.Generic;
+using System.Text.Json;
+
+namespace UnbrandedTypeSpec
+{
+    internal static partial class Optional
+    {
+        /// <param name="collection"> The collection. </param>
+        public static bool IsCollectionDefined<T>(IEnumerable<T> collection)
+        {
+            return !(collection is ChangeTrackingList<T> changeTrackingList && changeTrackingList.IsUndefined);
+        }
+
+        /// <param name="collection"> The collection. </param>
+        public static bool IsCollectionDefined<TKey, TValue>(IDictionary<TKey, TValue> collection)
+        {
+            return !(collection is ChangeTrackingDictionary<TKey, TValue> changeTrackingDictionary && changeTrackingDictionary.IsUndefined);
+        }
+
+        /// <param name="collection"> The value. </param>
+        public static bool IsCollectionDefined<TKey, TValue>(IReadOnlyDictionary<TKey, TValue> collection)
+        {
+            return !(collection is ChangeTrackingDictionary<TKey, TValue> changeTrackingDictionary && changeTrackingDictionary.IsUndefined);
+        }
+
+        /// <param name="value"> The value. </param>
+        public static bool IsDefined<T>(T? value)
+        where T : struct
+        {
+            return value.HasValue;
+        }
+
+        /// <param name="value"> The value. </param>
+        public static bool IsDefined(object value)
+        {
+            return value != null;
+        }
+
+        /// <param name="value"> The value. </param>
+        public static bool IsDefined(System.Text.Json.JsonElement value)
+        {
+            return value.ValueKind != System.Text.Json.JsonValueKind.Undefined;
```

Comment:
@jorgerangel-msft I believe you have added post processing to prevent inline namespaces. Do you happen to know why it's generating like this? Is there anything I'm missing?

Reply:
I'm not sure, I noticed this too hence why I opened https://github.com/microsoft/typespec/issues/3367 to investigate. We already added the `Reduce` functionality from Roslyn, so I'm guessing there is an issue with the metadata reference resolver event handler I added.

Reply:
Yeah we have this issue because Roslyn now does not know about anything in System.Text.Json dll